From 417bb1778799522ca623828ad4c5b0b99da59793 Mon Sep 17 00:00:00 2001
From: Tung Huynh <tung75605@gmail.com>
Date: Sat, 9 Dec 2023 01:32:50 -0800
Subject: [PATCH] direct3d11: use rect_dest for subpicture mapping

---
 modules/video_output/win32/direct3d11.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 9fe576ebad..e7762899ca 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -2179,8 +2179,8 @@ static int Direct3D11MapSubpicture(vout_display_t *vd, int *subpicture_region_co
         d3d_quad_t *quad = (d3d_quad_t *) quad_picture->p_sys;
 
         vout_display_cfg_t place_cfg = *vd->cfg;
-        place_cfg.display.width  = RECTWidth(sys->sys.rect_dest_clipped);
-        place_cfg.display.height = RECTHeight(sys->sys.rect_dest_clipped);
+        place_cfg.display.width  = RECTWidth(sys->sys.rect_dest);
+        place_cfg.display.height = RECTHeight(sys->sys.rect_dest);
         vout_display_place_t place;
         vout_display_PlacePicture(&place, &vd->source, &place_cfg, false);
 
-- 
2.42.0.windows.2

