From a1ede40b5c3d20bf201c4c8079452591013790dc Mon Sep 17 00:00:00 2001
From: Martin Finkel <me@martinfinkel.com>
Date: Mon, 25 Oct 2021 14:27:06 +0700
Subject: [PATCH] build: disable npapi/nsis/mix

---
 extras/package/win32/package.mak | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/extras/package/win32/package.mak b/extras/package/win32/package.mak
index b52acc697f..12334b3735 100644
--- a/extras/package/win32/package.mak
+++ b/extras/package/win32/package.mak
@@ -180,14 +180,8 @@ package-win32: package-win32-zip package-win32-7zip package-win32-exe package-wi
 
 package-win32-debug: package-win32-debug-zip package-win32-debug-7zip
 
-package-win32-release: package-win32-src $(win32_destdir)/NSIS/nsProcess.dll package-win-sdk
-	mkdir -p "$(win32_destdir)/msi/"
-	cp    $(top_builddir)/extras/package/win32/msi/config.wxi	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/axvlc.wxs		  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/bannrbmp.bmp	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/extensions.wxs	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/LICENSE.rtf	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/product.wxs	  "$(win32_destdir)/msi/"
+package-win32-release: package-win-strip package-win-sdk
+	7z a $(7Z_OPTS) $(WINVERSION)-release.7z $(win32_debugdir) "$(win32_destdir)/"
 
 	7z a $(7Z_OPTS) $(WINVERSION)-release.7z $(win32_debugdir) "$(win32_destdir)/"
 
-- 
2.47.3

