From 1b8e8c5cb5f587fdddd2ebb5477e64546d53dc92 Mon Sep 17 00:00:00 2001
From: Tung Huynh <tung75605@gmail.com>
Date: Sun, 4 May 2025 23:28:43 -0700
Subject: [PATCH] invoke appcontainer.pl

---
 extras/package/win32/build.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/extras/package/win32/build.sh b/extras/package/win32/build.sh
index bf709458aa..b724c45c4d 100755
--- a/extras/package/win32/build.sh
+++ b/extras/package/win32/build.sh
@@ -523,6 +523,9 @@ elif [ "$INSTALLER" = "r" ]; then
 make package-win32
 elif [ "$INSTALLER" = "u" ]; then
 make package-win32-release
+if [ ! -z "$WINSTORE" ]; then
+    find . \( -name "*.dll" -o -name "*.exe" \) -exec ../../patches/appcontainer.pl {} \;
+fi
 sha512sum vlc-*-release.7z
 elif [ -n "$INSTALL_PATH" ]; then
 make package-win-install
-- 
2.40.1

