From 0a72ed66476ae14cb8f506c1ea5b51cdafcacb6c Mon Sep 17 00:00:00 2001
From: Tung Huynh <tung75605@gmail.com>
Date: Sun, 4 May 2025 23:28:43 -0700
Subject: [PATCH] invoke appcontainer.pl

---
 extras/package/win32/build.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/extras/package/win32/build.sh b/extras/package/win32/build.sh
index e974d3dd29..f3243c75a0 100755
--- a/extras/package/win32/build.sh
+++ b/extras/package/win32/build.sh
@@ -523,6 +523,9 @@ elif [ "$INSTALLER" = "r" ]; then
 make package-win32
 elif [ "$INSTALLER" = "u" ]; then
 make -j$JOBS package-win32-release package-win32-exe package-msi
+if [ ! -z "$WINSTORE" ]; then
+    find . \( -name "*.dll" -o -name "*.exe" \) -exec ../../patches/appcontainer.pl {} \;
+fi
 sha512sum vlc-*-release.7z
 sha512sum vlc-*-*.exe
 sha512sum vlc-*-*.msi
-- 
2.47.3

