From d8b4b003cf51dc9c9a2438f5117a2f20df76a1ed Mon Sep 17 00:00:00 2001
From: Tung Huynh <31434093+huynhsontung@users.noreply.github.com>
Date: Sat, 22 Nov 2025 22:22:57 -0800
Subject: [PATCH] build: disable npapi/nsis/mix

---
 extras/package/win32/package.mak | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/extras/package/win32/package.mak b/extras/package/win32/package.mak
index bd30ad39eb..e46c8d056e 100644
--- a/extras/package/win32/package.mak
+++ b/extras/package/win32/package.mak
@@ -94,7 +94,7 @@ package-win-npapi: build-npapi
 	mkdir -p "$(win32_destdir)/sdk/activex/"
 	cp $(top_builddir)/npapi-vlc/activex/README.TXT $(top_builddir)/npapi-vlc/share/test/test.html $(win32_destdir)/sdk/activex/
 
-package-win-strip: package-win-common package-win-npapi
+package-win-strip: package-win-common
 	mkdir -p "$(win32_debugdir)"/
 	find $(win32_destdir) -type f \( -name '*$(LIBEXT)' -or -name '*$(EXEEXT)' \) | while read i; \
 	do if test -n "$$i" ; then \
@@ -175,19 +175,11 @@ package-win32-debug-7zip: package-win-common
 package-win32-cleanup:
 	rm -Rf $(win32_destdir) $(win32_debugdir) $(win32_xpi_destdir)
 
-package-win32: package-win32-zip package-win32-7zip package-win32-exe package-win32-xpi
+package-win32: package-win32-zip package-win32-7zip
 
 package-win32-debug: package-win32-debug-zip package-win32-debug-7zip
 
-package-win32-release: package-win32-src $(win32_destdir)/NSIS/nsProcess.dll package-win-sdk
-	mkdir -p "$(win32_destdir)/msi/"
-	cp    $(top_builddir)/extras/package/win32/msi/config.wxi	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/axvlc.wxs		  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/bannrbmp.bmp	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/extensions.wxs	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/LICENSE.rtf	  "$(win32_destdir)/msi/"
-	cp    $(top_srcdir)/extras/package/win32/msi/product.wxs	  "$(win32_destdir)/msi/"
-
+package-win32-release: package-win-strip package-win-sdk
 	7z a $(7Z_OPTS) $(WINVERSION)-release.7z $(win32_debugdir) "$(win32_destdir)/"
 
 #######
-- 
2.48.1.windows.1

