From 13835733a3a6b7b53127d2db81ebab5f2510a426 Mon Sep 17 00:00:00 2001
From: Tung Huynh <tung75605@gmail.com>
Date: Wed, 6 Dec 2023 15:24:07 -0800
Subject: [PATCH 3/3] fallback to vd->source

---
 modules/video_output/win32/common.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/video_output/win32/common.c b/modules/video_output/win32/common.c
index 4fdadec54f..c32d9c4171 100644
--- a/modules/video_output/win32/common.c
+++ b/modules/video_output/win32/common.c
@@ -162,6 +162,11 @@ void UpdateRects(vout_display_t *vd,
     bool is_forced)
 {
     vout_display_sys_t *sys = vd->sys;
+    msg_Dbg(vd, "sys->src_fmt: %p", sys->src_fmt);
+    if (!sys->src_fmt) {
+        msg_Dbg(vd, "sys->src_fmt is NULL. Initializing it to &vd->source");
+        sys->src_fmt = &vd->source;
+    }
     const video_format_t *source = sys->src_fmt;
 #define rect_src sys->rect_src
 #define rect_src_clipped sys->rect_src_clipped
-- 
2.42.0.windows.2

