From b834e12bf4b4cae2956d755d90df2f8f98dbb872 Mon Sep 17 00:00:00 2001
From: Tung Huynh <tung75605@gmail.com>
Date: Mon, 5 May 2025 00:54:01 -0700
Subject: [PATCH 1/2] get driver version for external direct3d11 device

---
 modules/video_output/win32/direct3d11.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 99d09b0511..f09e7ea27f 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -1735,6 +1735,8 @@ static int Direct3D11Open(vout_display_t *vd, bool external_device)
         }
 #endif
     }
+    else
+        D3D11_GetDriverVersion( vd, &sys->d3d_dev );
 
     IDXGISwapChain_QueryInterface( sys->dxgiswapChain, &IID_IDXGISwapChain4, (void **)&sys->dxgiswapChain4);
 
-- 
2.49.0.windows.1

