From 77457a84fa7846f53d5a1363d49e641b2b2b7e1e Mon Sep 17 00:00:00 2001
From: Bot <bot@example.com>
Date: Sat, 18 Jul 2026 00:53:37 -0700
Subject: [PATCH] fix SourceForge download links

---
 contrib/src/main.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index a4a930dd5c..f84b5a7390 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -19,7 +19,7 @@ VPATH := $(TARBALLS)
 
 # Common download locations
 GNU ?= http://ftpmirror.gnu.org/gnu
-SF := https://netcologne.dl.sourceforge.net/
+SF := https://downloads.sourceforge.net/project
 VIDEOLAN := http://downloads.videolan.org/pub/videolan
 CONTRIB_VIDEOLAN := http://downloads.videolan.org/pub/contrib
 GITHUB := https://github.com/
-- 
2.54.0.windows.1

